Fix building some samples in out of tree build

vadz opened the pull request:
See #23700.

vadz committed:
Fix building non-GUI samples out of tree

Define wxUSE_GUI=0 for them.

vadz committed:
Link OpenGL samples with GL library under Unix

This is not needed when building as part of the build tree, as it's
already part of EXTRALIBS_OPENGL then, but is required when building the
samples against installed libraries.

This commit is best viewed ignoring whitespace-only changes.

swt2c commented:
Thanks for the quick fix!  One note on the console fix, the archive sample's makefile.unx doesn't seem to have been updated, so might need a tweak for that.

vadz committed:
Don't use GUI settings for "archive" sample

This samples used "wx_append" in its bakefile, but should be using
"wx_append_base", as it's a console application.

This fixes its build with makefile.unx, as wxUSE_GUI wasn't properly
defined as 0 before.

vadz committed:
Compile Cocoa control example in Unix "widgets" sample makefile

Previously this was never done, as we couldn't determine if we were
using Cocoa or not at generation time, but we can determine it in the
makefile itself.

vadz committed:
Use flags required for using GTK in widgets sample

As we use GTK directly in this sample, we need to use the appropriate
compiler and link flags, so add them depending on the port in the
makefile.

vadz committed:
Exclude some bakefiles from Whitespace code check too

Add a couple of .bkl files containing TABs in the embedded make
fragments to the list of files skipped by the whitespace check.
